src/asound/bluealsa-ctl.c

```diff
@@ -992,23 +1120,90 @@ SND_CTL_PLUGIN_DEFINE_FUNC(bluealsa) {
 		goto fail;
 	}
 
-	if (!bluealsa_dbus_get_pcms(&ctl->dbus_ctx, &ctl->pcm_list, &ctl->pcm_list_size, &err)) {
+	if (!bluealsa_dbus_get_pcms(&ctl->dbus_ctx, &pcm_list, &pcm_list_size, &err)) {
 		SNDERR("Couldn't get BlueALSA PCM list: %s", err.message);
 		ret = -ENODEV;
 		goto fail;
 	}
 
+	if (bacmp(&device, BDADDR_ALL) != 0) {
```

Comment:
I think it will be easier to read if the if will be "true". More like `if (BDADDR_ALL) { ... }`
```suggestion
	if (bacmp(&device, BDADDR_ALL) == 0) {
		ctl->pcm_list = pcm_list;
		ctl->pcm_list_size = pcm_list_size;
	}
	else {
```